kernel/src/syscall/madvise.rs

```diff
@@ -20,11 +20,15 @@ pub fn sys_madvise(
     if start % PAGE_SIZE != 0 {
         return_errno_with_message!(Errno::EINVAL, "the start address should be page aligned");
     }
-    if len == 0 {
+    if len == 0 || len > usize::MAX - PAGE_SIZE + 1 {
         return Ok(SyscallReturn::Return(0));
```

Comment:
Are you sure that a too large value should be treated in the same way as `0`? This looks an `EINVAL` error to me.

Reply:
Yes, it should be EINVAL as https://elixir.bootlin.com/linux/v6.0.9/source/mm/madvise.c#L1399.
I misunderstand it as #L1407 condition...